File: company-service/src/main/java/com/food/delivery/companyservice/utils/properties/ActuatorProperties.java
package com.food.delivery.companyservice.utils.properties;

import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.ConstructorBinding;
import org.springframework.security.core.authority.SimpleGrantedAuthority;

@Getter
@AllArgsConstructor
@ConstructorBinding
@ConfigurationProperties("actuator")
public class ActuatorProperties {

	private final String path;
	private final String username;
	private final String password;
	private final SimpleGrantedAuthority authority;

	public String getAuthorityName() {
		return authority.getAuthority();
	}
}

File: menu-service/src/main/kotlin/com/food/delivery/menuservice/utils/properties/ActuatorProperties.kt
package com.food.delivery.menuservice.utils.properties

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding
import org.springframework.security.core.authority.SimpleGrantedAuthority

@ConstructorBinding
@ConfigurationProperties("actuator")
data class ActuatorProperties(val path: String, val username: String, val password: String,
							  val authority: SimpleGrantedAuthority) {

	fun getAuthorityName(): String {
		return authority.authority
	}
}
